src/main/java/com/appleaster/command/Command.java

```diff
@@ -0,0 +1,56 @@
+package com.appleaster.command;
+import java.time.LocalDate;
+
+import com.appleaster.task.Task;
+
+public class Command {
+    private final CommandType type;
+    private Task task;
+    private int taskIndex;
+    private LocalDate date;
+    private String keyword;    
+
+    public Command(CommandType type) {
+        this.type = type;
+    }
+
+    public Command(CommandType type, String keyword) {
+        this.type = type;
+        this.keyword = keyword;
+    }    
+
+    public Command(CommandType type, Task task) {
+        this.type = type;
+        this.task = task;
+    }
+
+    public Command(CommandType type, int taskIndex) {
+        this.type = type;
+        this.taskIndex = taskIndex;
+    }
+
+    public Command(CommandType type, LocalDate date) {
+        this.type = type;
+        this.date = date;
+    }
+
+    public CommandType getType() {
+        return type;
+    }
+
+    public Task getTask() {
```

Comment:
# Potential NullPointerException
- Since your overloaded constructors do not initialise the values of the other fields, only specific ones.
- Calling upon getTask() on a Command object constructed with public Command(CommandType type) might throw a NullPointerException.
- Consider including default values for the other fields if not in use to prevent this.